[{"count": 0, "tags": [], "text": "With a certain type of XLS data I get this exception:\n\nException in thread \"main\" org.apache.poi.hssf.record.RecordFormatException: Unable to construct record instance\n\tat org.apache.poi.hssf.record.RecordFactory$ReflectionRecordCreator.create(RecordFactory.java:71)\n\tat org.apache.poi.hssf.record.RecordFactory.createSingleRecord(RecordFactory.java:254)\n\tat org.apache.poi.hssf.record.RecordFactory.createRecords(RecordFactory.java:373)\n\tat org.apache.poi.hssf.usermodel.HSSFWorkbook.<init>(HSSFWorkbook.java:277)\n\tat org.apache.poi.hssf.usermodel.HSSFWorkbook.<init>(HSSFWorkbook.java:202)\n\tat org.apache.poi.hssf.usermodel.HSSFWorkbook.<init>(HSSFWorkbook.java:184)\n\tat ExcelExtractor.openWorkbook(ExcelExtractor.java:668)\n\tat ExtractorTest.main(ExtractorTest.java:70)\nCaused by: org.apache.poi.hssf.record.RecordFormatException: Not enough data (0) to read requested (2) bytes\n\tat org.apache.poi.hssf.record.RecordInputStream.checkRecordPosition(RecordInputStream.java:185)\n\tat org.apache.poi.hssf.record.RecordInputStream.readUShort(RecordInputStream.java:230)\n\tat org.apache.poi.hssf.record.pivottable.ExtendedPivotTableViewFieldsRecord.<init>(ExtendedPivotTableViewFieldsRecord.java:53)\n\tat sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)\n\tat sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:39)\n\tat sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)\n\tat java.lang.reflect.Constructor.newInstance(Constructor.java:494)\n\tat org.apache.poi.hssf.record.RecordFactory$ReflectionRecordCreator.create(RecordFactory.java:63)\n\t... 7 more\n\n\nExcel has no problem opening this file, POI 3.1 works fine on the input.\nI checked with POI 3.5 latest svn March 26.", "attachment_id": null, "id": 125803, "creator": "axel.roeslein@googlemail.com", "time": "2009-03-26T04:07:29Z", "bug_id": 46918, "creation_time": "2009-03-26T04:07:29Z", "is_private": false}, {"attachment_id": null, "tags": [], "bug_id": 46918, "is_private": false, "count": 1, "id": 125807, "time": "2009-03-26T05:05:31Z", "creator": "apache@gagravarr.org", "creation_time": "2009-03-26T05:05:31Z", "text": "Please upload a sample problem file, so we can take a look at why some records are the wrong length"}, {"count": 2, "tags": [], "creator": "axel.roeslein@googlemail.com", "is_private": false, "id": 125811, "creation_time": "2009-03-26T05:50:01Z", "time": "2009-03-26T05:50:01Z", "bug_id": 46918, "text": "When checking the file in question BiffViewer throws an Exception\n\njava -cp build/classes org.apache.poi.hssf.dev.BiffViewer /tmp/t1.xls > /tmp/t1.txt\njava.lang.ArrayIndexOutOfBoundsException\n        at java.lang.System.arraycopy(Native Method)\n        at org.apache.poi.ddf.UnknownEscherRecord.fillFields(UnknownEscherRecord.java:77)\n        at org.apache.poi.ddf.EscherContainerRecord.fillFields(EscherContainerRecord.java:53)\n        at org.apache.poi.ddf.EscherContainerRecord.fillFields(EscherContainerRecord.java:53)\n        at org.apache.poi.hssf.record.AbstractEscherHolderRecord.convertToEscherRecords(AbstractEscherHolderRecord.java:82)\n        at org.apache.poi.hssf.record.AbstractEscherHolderRecord.<init>(AbstractEscherHolderRecord.java:68)\n        at org.apache.poi.hssf.record.DrawingGroupRecord.<init>(DrawingGroupRecord.java:41)\n        at org.apache.poi.hssf.dev.BiffViewer.createRecord(BiffViewer.java:149)\n        at org.apache.poi.hssf.dev.BiffViewer.createRecords(BiffViewer.java:84)\n        at org.apache.poi.hssf.dev.BiffViewer.main(BiffViewer.java:398)\n\n\nThe output looks ok to me.\nWhat should I check within the BiffViewer output?\n(content cannot be made public)", "attachment_id": null}, {"count": 3, "tags": [], "creator": "apache@gagravarr.org", "attachment_id": null, "id": 125818, "creation_time": "2009-03-26T08:29:11Z", "time": "2009-03-26T08:29:11Z", "bug_id": 46918, "text": "Looking at your stacktrace, your exception is apparently being triggered by some invalid escher record headers. However, without the file we can't really do much, so alas you'll need to dive into the poi sourcecode in your debugger and investigate it yourself (well, or find a file you can share that also triggers the issue!)", "is_private": false}, {"count": 4, "tags": [], "bug_id": 46918, "attachment_id": null, "id": 125821, "time": "2009-03-26T09:03:43Z", "creator": "axel.roeslein@googlemail.com", "creation_time": "2009-03-26T09:03:43Z", "is_private": false, "text": "I wish I could debug.\n\nUsing Eclipse 3.5, creating a new project, importing from existing source, using defaults\n\n106 errors, e.g.\nNumericRanges cannot be resolved\tColumnHelper.java\tPOI35/ooxml/java/org/apache/poi/xssf/usermodel/helpers\tline 176\tJava Problem\nNumericRanges cannot be resolved\tTestNumericRanges.java\tPOI35/ooxml/testcases/org/apache/poi/xssf/util\tline 55\tJava Problem\n\n\nIt might of course be caused by my Eclipse settings.\nIs there any adaptable .classpath available for a quick start with Eclipse?\n\nRegards,\nAxel."}, {"attachment_id": null, "tags": [], "bug_id": 46918, "is_private": false, "count": 5, "id": 125920, "time": "2009-03-31T14:18:42Z", "creator": "josh@apache.org", "creation_time": "2009-03-31T14:18:42Z", "text": "Axel supplied me with the hex of the offending ExtendedPivotTableViewFieldsRecord:\n00 01 0A 00 1E 14 00 0A FF FF FF FF 00 00\n\nThis is 10 bytes shorter than POI was expecting (existing test cases have 20 bytes of data in this record).  The 10 byte format is probably from earlier Excel versions (perhaps 97) POI now reads both formats of this record.\n\nFixed in svn r760464\n\njunit added"}]